code/modules/implant_hacking/_hacking.dm

```diff
@@ -0,0 +1,162 @@
+
+	game_list = list()
+
+	var/diffrences = 3
+	current_user = user
+	cybernetic = target
+
+	for(var/info in cybernetic.encode_info)
+		if(cybernetic.encode_info[info] == 0)
+			continue
+		var/list/encrypted_information = cybernetic.encode_info[info]
+
+		for(var/protocol in encrypted_information)
+			if(protocol in parent_cyberlink.encode_info[info])
+				diffrences--
+				break
+
+	if(diffrences == 0)
+		to_chat(current_user,"<span class='notice'> Cyberlink beeps: [uppertext(cybernetic.name)] ALREADY COMPATIBLE.</span>")
+		cleanup()
+		return
+
+	if(!game_list.len)
+		for(var/i in 1 to diffrences)
+			var/datum/hacking_minigame/game = new/datum/hacking_minigame()
+			game.generate()
+			game_list += game
+
+	ui_interact(user)
+
+/obj/item/cyberlink_connector/proc/cleanup()
+	current_user = null
+	cybernetic = null
+	QDEL_LIST(game_list)
+	processing = FALSE
+
+/obj/item/cyberlink_connector/proc/hack_success(success as num)
+	for(var/info in cybernetic.encode_info)
+		if(cybernetic.encode_info[info] == 0)
+			continue
+		//Not a += because we want to avoid having duplicate entries in either encode_info
+		cybernetic.encode_info[info] |= parent_cyberlink.encode_info[info]
+	current_user.mind.adjust_experience(/datum/skill/implant_hacking,success * 25)
+	to_chat(current_user,"<span class='notice'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] SUCCESS. COMPATIBILITY ACHIEVED.</span>")
+	cleanup()
+
+/obj/item/cyberlink_connector/proc/hack_failure(failed as num)
+	var/chance = rand(0,25*failed)
+	switch(chance)
+		if(0 to 25)
+			to_chat(current_user,"<span class='warning'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] MINOR FAILURE. COMPATIBILITY NOT ACHIEVED. NO DAMAGE DETECTED.</span>")
+		if(26 to 40)
+			to_chat(current_user,"<span class='warning'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] MEDIUM FAILURE. COMPATIBILITY NOT ACHIEVED. SMALL AMOUNT OF DAMAGE DETECTED.</span>")
+			current_user.adjustFireLoss(10)
+			current_user.emote("scream")
+		if(41 to 50)
+			to_chat(current_user,"<span class='warning'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] MEDIUM FAILURE. COMPATIBILITY NOT ACHIEVED. PROTOCOL SCRAMBILING DETECTED.</span>")
+			cybernetic.random_encode()
+		if(51 to 75)
+			to_chat(current_user,"<span class='danger'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] MAJOR FAILURE. COMPATIBILITY NOT ACHIEVED. MINOR ELECTROMAGNETIC PULSE DETECTED.</span>")
+			empulse(current_user, 0, 1)
+		if(76 to 99)
+			to_chat(current_user,"<span class='danger'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] MAJOR FAILURE. COMPATIBILITY NOT ACHIEVED. MAJOR ELECTROMAGNETIC PULSE DETECTED.</span>")
+			empulse(current_user, 1, 2)
+		if(100)
+			to_chat(current_user,"<span class='danger'> Cyberlink beeps: HACKING [uppertext(cybernetic.name)] CRITICAL FAILURE. COMPATIBILITY NOT ACHIEVED. IMPLANT OVERHEATING IN 5 SECONDS.</span>")
+			addtimer(CALLBACK(src, .proc/explode), 5 SECONDS)
```

Comment:
Make it output some sound if you're not gonna make this a visible message

Reply:
oh yeah right visible message. forgot about adding that.